### DISCUSSION TITLE
ZenMode Brainstorming

### DISCUSSION THREAD

**Discussion by gibru** [nunocoracao/blowfish#1155] [Ideas]
Hey @v20100v

In reply to your message, first a thank you for this cool feature! I'm currently in the process of migrating to Blowfish and your feature is something I didn't know I needed.

At the moment, I don't have better ideas, but maybe we get inspired? To that end, I've been using and playing with ZenMode to see how I can integrate it into my website and one cool thing would be an on/off toggle anywhere on the page. That's especially useful with long-form content as scrolling/jumping to the top breaks the reading flow.

Right now, there's the scroll-to-top button that's always visible (if set to true I guess) so to simply put the toggle next to it would look a bit weird I guess. How about a frame around the scroll-to-top button? Just a thin, unobtrusive one. That way, when "showZenMode" is set to true, the frame expands allowing for the ZenMode toggle to be next to or below/above the scroll-to-top button and it looks as if it were a unified place for scroll-top and ZenMode.

Another thing I really like is the new width. Widescreen mode? This is something that for some of my publications is extremely useful. For instance, when all I publish is a picture. Here, it'd be cool to have ZenMode's widescreen mode independently of ZenMode. In other words, while it makes sense to have a ZenMode toggle to remove all distractions including author profile, hero stuff etc., WideScreenMode would be something that could be set in the front matter to either true/false to set it as a default on a per article basis.

Thoughts?

**Comment by v20100v** [comment]
Hello @gibru,

> *In reply to your message, first a thank you for this cool feature! I'm currently in the process of migrating to Blowfish and your feature is something I didn't know I needed.*

Thank you for your message. I'm like you, migrating to SSG Huho with Blowfish.

---

> *one cool thing would be an on/off toggle anywhere on the page. This is especially useful with long content, as scrolling/jumping up interrupts the reading flow (...)*

You are quite right. When the content of the page is long, it is not pleasant (user friendly) to have to scroll. We could imagine an option that set the position of the toogle zen-mode button in the page. Could you make a (simple) drawing to position this button ?

---

> Another thing I really like is the new width. Widescreen mode? This is something that is extremely useful for some of my publications. For example, when I only publish one image. Here it would be cool to have ZenMode's widescreen mode independent of ZenMode. In other words, while it makes sense to have a ZenMode toggle to remove all distractions, including author profile, hero items, etc., WideScreenMode would be something that could be set right away. start on true/false to set it by default to per item.

Same, would it be possible to have a drawing of your idea for WideScreenMode? It seems to correspond to the first version that I submitted for discussion with @nunocoracao. He had not retained the total occupancy of the width of the screen. That’s debatable, but at the end, the arbitration belongs to the owner of this project ^^.

Regards.

**Comment by gibru** [comment]
> Could you make a (simple) drawing to position this button ?

Very simple: let's say we just frame the existing to-top button like so:

![without_zen](https://github.com/nunocoracao/blowfish/assets/14093742/3207eb29-7c16-4df0-a46a-7d68dd9a357c)

Next, assuming we have set `showZenMode:true` in the front matter, we could make the ZenMode toggle appear next to the to-top botton like so:

![with_zen](https://github.com/nunocoracao/blowfish/assets/14093742/97a06d10-062b-4080-9b95-f9d257f1e805)

Same if to-top is deactivated. There's just a framed ZenMode toggle.

Of course, in case there's a worry that this might be confused as one toggle, we could put them in their own separate squares? Just as an idea to get the ball rolling.

> would it be possible to have a drawing of your idea for WideScreenMode?

Well, your existing solution is already ideal as far as I'm concerned. I would simply "decouple" the WideScreenMode so that it can be set independently from ZenMode in the front matter. A simple `showWidescreenMode:true` would be enough I guess?
